ydb/core/grpc_services/rpc_object_storage.cpp

```diff
@@ -820,13 +820,23 @@ class TObjectStorageListingRequestGrpc : public TActorBootstrapped<TObjectStorag
         };
 
         const auto getTypeFromColMeta = [&](const auto &colMeta) {
-            if (colMeta.PType.GetTypeId() == NScheme::NTypeIds::Pg) {
+            const NScheme::TTypeInfo& typeInfo = colMeta.PType;
+            switch (typeInfo.GetTypeId()) {
+            case NScheme::NTypeIds::Pg: {
                 return NYdb::TTypeBuilder().Pg(getPgTypeFromColMeta(colMeta)).Build();
-            } else {
+            }
+            case NScheme::NTypeIds::Decimal: {
+                return NYdb::TTypeBuilder().Decimal(NYdb::TDecimalType(
+                        typeInfo.GetDecimalType().GetPrecision(), 
+                        typeInfo.GetDecimalType().GetScale()))
+                    .Build();
+            }
+            default :{
```

Comment:
NIT:
maybe format?
also scope for these cases in this switch are unnecessary

Reply:
0b2b82fab577a42e687f3e1d0b7ea3dbd5e26c32